src/structs/API.ts

```diff
@@ -0,0 +1,314 @@
+import fetch, { Headers } from 'node-fetch'
+import APIError from '../utils/APIError'
+import ClientUtils, { ClientType } from '../utils/ClientUtils'
+import qs from 'querystring'
+import '../typings'
+import { EventEmitter } from 'events'
+
+interface APIOptions {
+  /**
+   * Top.GG Token
+   */
+  token: any
+  /**
+   * Interval at which to post in milliseconds
+   * @default 1800000 30 minutes
+   */
+  interval?: number
+  /**
+   * Whether or not to autopost
+   * @default true
+   */
+  autoPost?: boolean
+  /**
+   * Whether or not in testing phase (devs only)
+   */
+  debug?: boolean
+}
+
+/**
+ * Top.GG API Client for Posting stats or Fetching data
+ * @link https://top.gg/api/docs
+ * @example
+ * ```js
+ * const client = Discord.Client() // Your discord.js or Eris client
+ *
+ * const TopGG = require('@top-gg/sdk')
+ *
+ * const api = new TopGG.API('Your top.gg token', client)
+ *
+ * api.on('posted', () => {
+ *   console.log('Server count posted!')
+ * })
+ * ```
+ */
+export default class API extends EventEmitter {
+  private options: APIOptions
+  private client: any
+  private clientType: ClientType
+  /**
+   * Create Top.GG API instance
+   * @param token Token or options
+   * @param client Supported library client 
+   */
+  constructor (token?: string|APIOptions, client?: any) {
+    super()
+    const defaultOptions = {
+      interval: 1800000,
+      autoPost: true,
+      debug: false
+    }
+    if (!token || typeof token === 'string') {
+      this.options = {
+        token,
```

Comment:
might not need to be an if statement actually.

Reply:
https://github.com/top-gg/dblapi.js/pull/35#discussion_r534349591